packages/scm/src/browser/scm-widget.tsx

```diff
@@ -171,4 +171,23 @@ export class ScmWidget extends BaseWidget implements StatefulWidget {
         this.resourceWidget.restoreState(changesTreeState);
     }
 
+    collapseScmTree(): void {
+        const { model } = this.resourceWidget;
+        const root = model.root;
+        if (CompositeTreeNode.is(root)) {
+            root.children.map(group => {
+                if (CompositeTreeNode.is(group)) {
+                    group.children.map(folderNode => {
+                        if (CompositeTreeNode.is(folderNode)) {
+                            model.collapseAll(folderNode);
+                        }
+                        if (SelectableTreeNode.isSelected(folderNode)) {
+                            folderNode.selected = false;
```

Comment:
you should not mutate nodes yourself, it should happen in `TreeSelectionService`, you should call  `this.model.selectNode`  instead

Reply:
@akosyakov When I tested the previous changes, I noticed that the last composite node was selected when the `collapseAll` button is clicked. 

However, I removed that part of the code since the last node was not selected when I tested the changes today.

Reply:
If we do not fix the selection, we end up in a state where the last composite node (collapsable node) is selected.
I believe the correct behavior would be to select the first composite node (or select the hidden root like vscode):


_After Performing a Collapse-All_:

<div align='left'>

<img width="498" alt="Screen Shot 2020-08-07 at 7 58 56 PM" src="https://user-images.githubusercontent.com/40359487/89697255-5e579880-d8e9-11ea-94f2-1520dfb00c9b.png">

</div>

<br />

It is reproducible when the changes including multiple composite nodes.

Reply:
ok, then call `model.selectNode`. Nodes are immutable, to mutate the state you should always use the model, it will update internal and node state properly and notify all views. Mutating nodes directly won't do any good, there is no even guarantee that a node still belongs to the model, when you call some model method it will under the hood find a good node if it exists.